src/Microsoft.Data.Sqlite/SqliteConnection.cs

```diff
@@ -105,52 +111,76 @@ public override void Open()
             SetFolders();
         }
 
-#if !NETCORE50
-        private void SetFolders() { }
+        partial void SetFolders();
 
-        private string AdjustForRelativeDirectory(string path)
-            => path;
-#else
-        private string AdjustForRelativeDirectory(string path)
+#if NETCORE50
+        partial void SetFolders()
         {
-            var appData = GetApplicationData();
-            try
+            if (TemporaryFolderPath != null)
             {
-                if (appData == null || Path.IsPathRooted(path))
-                {
-                    return path;
-                }
-
-                return Path.GetFullPath(Path.Combine(appData.LocalFolder.Path, path));
+                DbConnectionExtensions.ExecuteNonQuery(this, "PRAGMA temp_store_directory = '" + TemporaryFolderPath + "'");
             }
-            catch (NotSupportedException)
+        }
+        
+        private readonly static string DefaultBasePath = AppData?.LocalFolder.Path;
+        private readonly static string TemporaryFolderPath = AppData?.TemporaryFolder.Path;
+
+        private static dynamic AppData
+        {
+            get
             {
-                Debug.WriteLine("Could not adjust relative path for use on UWP.");
+                var appDataType = Type.GetType("Windows.Storage.ApplicationData, Windows, ContentType=WindowsRuntime", throwOnError: false);
+                var appData = (dynamic)appDataType?.GetTypeInfo()
+                    .GetDeclaredProperty("Current").GetMethod.Invoke(null, null);
+                return appData;
             }
-            return path;
         }
+#else
+        private readonly static string DefaultBasePath = PlatformServices.Default.Application.ApplicationBasePath;
+#endif
 
-        private void SetFolders()
+        internal static string AdjustForRelativeDirectory(string path)
         {
-            var appData = GetApplicationData();
+            if (DefaultBasePath == null)
+            {
+                return path;
+            }
 
-            if (appData == null)
+            if (path.StartsWith("file:"))
             {
-                return;
+#if NETCORE50
+                // UWP cannot adjust for URI on UWP
+                return path;
```

Comment:
Named, shared-cache, in-memory databases are the only scenarios I find interesting here. :smile_cat: